docs/madr/decisions/070-resource-identifier.md

```diff
@@ -0,0 +1,363 @@
+| Listener         | `kuma:dns`<br>`kuma:envoy:admin`<br>`kuma:metrics:prometheus`<br>`_kuma:metrics:opentelemetry:<backendName>`                                                         | –                    |
+| VirtualHost      | listener's name                                                                                                                                                      | –                    |
+| Internal Cluster | `kuma:readiness`<br>`kuma:envoy:admin`<br>`kuma:metrics:hijacker`<br>`_kuma:metrics:opentelemetry:<backendName>`<br>`tracing:`<br>`access_log_sink`<br>`ads_cluster` | –                    |
+
+#### Envoy stats
+
+There are 2 fields to specify alternative name for emitting stats – `stat_prefix`, `alt_stat_name`.
+Field `alt_stat_name` is used only for Cluster, and it's automatically converts all `:` to `_` by Envoy when emitting statistics.
+Field `stat_prefix` is used for HTTPConnectionManager, TCPProxy, KafkaBroker (network filter) and RBAC (network filter).
+Unlike `alt_stat_name` the field `stat_prefix` doesn't have any restrictions from Envoy.
+Kuma sanitizes both fields using `SanitizeMetric` function before setting them on Envoy's resource:
+
+```go
+var illegalChars = regexp.MustCompile(`[^a-zA-Z_\-0-9]`)
+
+// We need to sanitize metrics in order to  not break statsd and prometheus format.
+// StatsD only allow [a-zA-Z_\-0-9.] characters, everything else is removed
+// Extra dots breaks many regexes that converts statsd metric to prometheus one with tags
+func SanitizeMetric(metric string) string {
+	return illegalChars.ReplaceAllString(metric, "_")
+}
+```
+
+I can't find explanation why `alt_stat_name` and `stat_prefix` are treated differently by Envoy. 
+When Envoy exports metrics in Prometheus format resource name is always emitted as label value, it's never part of the metric name, i.e:
+```
+envoy_cluster_upstream_cx_total{envoy_cluster_name="localhost_5000"} 0
+```
+so there is no reason to replace `:` with `_`.
+
+If Envoy tried to support StatsD format then there are still a question, why `stat_prefix` emits metrics with `:` but `alt_stat_name` replaces `:` with `_`.
+Also, StatsD format doesn't seem to be supported out-of-the-box by Envoy and requires configuring `stats_sinks` in the static config.
+
+### Constraints
+
+#### What characters we can use in the identifier?
+ 
+In URL path segment we can use:
+
+```
+ALPHA / DIGIT / "-" / "." / "_" / "~" / "!" / "$" / "&" / "'" / "(" / ")" / "*" / "+" / "," / ";" / "=" / ":" / "@"
+```
+
+In Envoy resource names we can use any character except `:` if we want to solve [Issue #2363](https://github.com/kumahq/kuma/issues/2363) without workaround.
+
+In Envoy stats fields we can use any character except `:` to avoid on the fly conversion of `:` to `_` for Cluster names in Prometheus labels.
```

Comment:
```suggestion
In Envoy stats fields we can use any character except `:` to avoid on the fly conversion of `:` to `_` for Cluster names in Prometheus label values.
```

Labels have keys and values it's worth being thourough